synapse/module_api/__init__.py

```diff
@@ -328,6 +332,7 @@ def register_password_auth_provider_callbacks(
             is_3pid_allowed=is_3pid_allowed,
             auth_checkers=auth_checkers,
             get_username_for_registration=get_username_for_registration,
+            get_display_name_for_registration=get_display_name_for_registration,
```

Comment:
Do we want to note in the docstring that a new property is being added?

Reply:
I think the mention of the Synapse version in the callback doc should be good enough?

Reply:
But the docstring is wrong now, it says it was introduced in a certain version, but that's only partially true?

Reply:
I wonder if we should just remove the "Introduced in..." mentions for the register callbacks functions, given we already track that info for individual callbacks in the docs it sounds more confusing than useful.

Reply:
I've opened https://github.com/matrix-org/synapse/issues/12017 to track this (since it's not exclusive to auth provider callbacks).